tests/test_edge_cases.py

```diff
@@ -1427,31 +1512,35 @@ def validator(v):
         yield validator
 
 
-@pytest.mark.xfail(reason='working on V2')
 @pytest.mark.parametrize(
     'type_,expected',
     [
         (int, 'int'),
-        (Optional[int], 'Optional[int]'),
+        (Optional[int], 'Union[int, NoneType]'),
         (Union[None, int, str], 'Union[NoneType, int, str]'),
         (Union[int, str, bytes], 'Union[int, str, bytes]'),
         (List[int], 'List[int]'),
         (Tuple[int, str, bytes], 'Tuple[int, str, bytes]'),
         (Union[List[int], Set[bytes]], 'Union[List[int], Set[bytes]]'),
         (List[Tuple[int, int]], 'List[Tuple[int, int]]'),
-        (Dict[int, str], 'Mapping[int, str]'),
+        (Dict[int, str], 'Dict[int, str]'),
         (FrozenSet[int], 'FrozenSet[int]'),
         (Tuple[int, ...], 'Tuple[int, ...]'),
-        (Optional[List[int]], 'Optional[List[int]]'),
+        (Optional[List[int]], 'Union[List[int], NoneType]'),
         (dict, 'dict'),
-        (DisplayGen[bool, str], 'DisplayGen[bool, str]'),
+        (DisplayGen[bool, str], ('DisplayGen[bool, str]', 'tests.test_edge_cases.DisplayGen[bool, str]')),
```

Comment:
Weirdly the display is different in Python<=3.9 and Python>=3.10, which is why I added two cases here.

Happy to change it back to only having one value and ensuring it's the same across python versions if that's preferable (or at least adding a case that differs for different `sys.version_info` values)

Reply:
i think it's fine for it to be different, but if i'm being nitpicky about the tests, better to do

```
pytest.param(DisplayGen[bool, str], 'DisplayGen[bool, str]', mark=pytest.mark.skipif(sys.version_info < ...))),
```

And then keep the function body as-was.